optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/list/OriginalListChangeIterator.java

```diff
@@ -80,6 +81,14 @@ public OriginalListChangeIterator(
             destinationIndexIterator = listIndexIterator(upcomingDestinationEntity);
         }
 
+        if (upcomingSourceEntity == null && upcomingSourceIndex == null) {
+            return new ListAssignMove<>(
+                    listVariableDescriptor,
+                    upcomingValue,
+                    upcomingDestinationEntity,
+                    destinationIndexIterator.nextInt());
+        }
```

Comment:
It's interesting that there are no unassigned move created in this iterator. It makes sense (unless nullable=true which we don't support yet), so this is good.
Maybe we can add a comment about that. I took me a while to realize this is not a bug.

```suggestion
        }
        // No need to generate ListUnassignMove because they are only used as undo moves.
```